warp10/src/main/sh/warp10.sh

```diff
@@ -34,6 +41,12 @@ fi
 JAVA_OPTS=${JAVA_OPTS:-}
 set -euo pipefail
 
+# run warp10-env.sh, when the file exists, to predefine user variables
+BIN_DIR=$(dirname $0)
+if [[ -e "${BIN_DIR}/warp10-env.sh" ]]; then
+  source "${BIN_DIR}/warp10-env.sh"
```

Comment:
loading warp10-env.sh here will clear any existing JAVA_OPTS, hence my suggestion to use JAVA_EXTRA_OPTS instead of JAVA_OPTS